synapse/handlers/presence.py

```diff
@@ -979,14 +981,15 @@ def should_notify(old_state, new_state):
     return False
 
 
-def format_user_presence_state(state, now):
+def format_user_presence_state(state, now, include_user_id=True):
     """Convert UserPresenceState to a format that can be sent down to clients
     and to other servers.
```

Comment:
Maybe add a comment to explain why the "user_id" is optional? something like
```
The "user_id" is optional so that this function can be used to format presence 
updates for client /sync responses and for federation /send requests.
```